06_neighborhoods/Transportation/transit_trips_cost_county_subgroups.R

```diff
@@ -0,0 +1,645 @@
+      perc_POC >= 0.6 ~ "Majority Non-White Tracts",
+      perc_white >= 0.4 | perc_POC >= 0.4 ~ "Mixed Race and Ethnicity Tracts",
+      perc_white < 0.4 & perc_POC < 0.4 ~ as.character(NA)
+    ))
+
+cost_data_2019 <- cost_data_2019 %>%
+  mutate(
+    race_category = case_when(
+      is.na(perc_white) | is.na(perc_POC) ~ as.character(NA), 
+      perc_white >= 0.6 ~ "Majority White-NH Tracts",
+      perc_POC >= 0.6 ~ "Majority Non-White Tracts",
+      perc_white >= 0.4 | perc_POC >= 0.4 ~ "Mixed Race and Ethnicity Tracts",
+      perc_white < 0.4 & perc_POC < 0.4 ~ as.character(NA)
+    ))
+
+trips_data_2015 <- trips_data_2015 %>%
+  mutate(
+    race_category = case_when(
+      is.na(perc_white) | is.na(perc_POC) ~ as.character(NA), 
+      perc_white >= 0.6 ~ "Majority White-NH Tracts",
+      perc_POC >= 0.6 ~ "Majority Non-White Tracts",
+      perc_white >= 0.4 | perc_POC >= 0.4 ~ "Mixed Race and Ethnicity Tracts",
+      perc_white < 0.4 & perc_POC < 0.4 ~ as.character(NA)
+    ))
+
+trips_data_2019 <- trips_data_2019 %>%
+  mutate(
+    race_category = case_when(
+      is.na(perc_white) | is.na(perc_POC) ~ as.character(NA), 
+      perc_white >= 0.6 ~ "Majority White-NH Tracts",
+      perc_POC >= 0.6 ~ "Majority Non-White Tracts",
+      perc_white >= 0.4 | perc_POC >= 0.4 ~ "Mixed Race and Ethnicity Tracts",
+      perc_white < 0.4 & perc_POC < 0.4 ~ as.character(NA)
+    ))
+
+############################################################################
+############################################################################
+############################################################################
+##########################################################################
+###################################################################
+
+# 3a. Collapse to the county level to create ALL values (to be appended later)
+
+cost_all_15 <- cost_data_2015 %>%
+  group_by(state, county) %>%
+  summarize(index_transportation_cost = weighted.mean(x = t_80ami, w = households, na.rm = TRUE)) %>%
+  mutate(subgroup_type = "race-ethnicity",
+         subgroup = "All")
+
+cost_all_19 <- cost_data_2019 %>%
+  group_by(state, county) %>%
+  summarize(index_transportation_cost = weighted.mean(x = t_80ami, w = households, na.rm = TRUE)) %>%
+  mutate(subgroup_type = "race-ethnicity",
+         subgroup = "All")
+
+trips_all_15 <- trips_data_2015 %>%
+  group_by(state, county) %>%
+  summarize(index_transit_trips = weighted.mean(x = transit_trips_80ami, w = households, na.rm = TRUE)) %>%
+  mutate(subgroup_type = "race-ethnicity",
+         subgroup = "All")
+
+trips_all_19 <- trips_data_2019 %>%
+  group_by(state, county) %>%
+  summarize(index_transit_trips = weighted.mean(x = transit_trips_80ami, w = households, na.rm = TRUE)) %>%
+  mutate(subgroup_type = "race-ethnicity",
+         subgroup = "All")
+
+# Now make sure to change the trips to national percentile ranks
+trips_all_15$rank <- rank(trips_all_15$index_transit_trips)
```

Comment:
When I run this code it appears NAs are getting ranked. Is that right? It doesn't seem right to me.

Reply:
Updated to calculate ranks for only non-NA values (rows 370-380)